contracts/eos/BancorConverterRegistry/BancorConverterRegistry.cpp

```diff
@@ -0,0 +1,165 @@
+#include "../Common/common.hpp"
+#include "../Token/Token.hpp"
+#include "../BancorConverter/BancorConverter.hpp"
+#include "BancorConverterRegistry.hpp"
+
+ACTION BancorConverterRegistry::addconverter(const name& converter_account, const symbol_code& pool_token_code) {
+    const extended_symbol pool_token = get_converter_pool_token(converter_account, pool_token_code);
+    const converter_t converter = converter_t{ converter_account, pool_token };
+    check(is_converter_active(converter), "converter is inactive");
+
+    register_converter(converter);
+    register_pool_token(converter.pool_token);
+    register_convertible_pairs(converter);
+
+    BancorConverter::reserves reserves_table(converter.contract, converter.pool_token.get_symbol().code().raw());
+    uint64_t num_of_reserves = std::distance(reserves_table.cbegin(), reserves_table.cend());
+    if (num_of_reserves > 1)
+        register_liquidity_pool(converter);
+}
+
+ACTION BancorConverterRegistry::rmconverter(const name& converter_account, const symbol_code& pool_token_code) {
+    const extended_symbol pool_token = get_converter_pool_token(converter_account, pool_token_code);
+    const converter_t converter = converter_t{ converter_account, pool_token };
+    check(has_auth(get_self()) || !is_converter_active(converter), "cannot remove an active converter");
+
+    unregister_converter(converter);
+    unregister_pool_token(converter.pool_token);
+    unregister_convertible_pairs(converter);
+    unregister_liquidity_pool(converter);
+}
+
+void BancorConverterRegistry::register_converter(const converter_t& converter) {
+    converters converters_table(get_self(), converter.pool_token.get_symbol().code().raw());
+    const auto p_converter = converters_table.find(converter.pool_token.get_contract().value);
+    check(p_converter == converters_table.end(), "converter already exists");
+    converters_table.emplace(get_self(), [&converter](converter_t& c){
+        c = converter;
```

Comment:
if we keep the converter here and not the pool tokens, how does a DAPP get a list of all the pool tokens (with reserve count > 1)?

Reply:
the `liqdtypools` table stores all the converters with more than 1 reserve